src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java

```diff
@@ -976,8 +1049,51 @@ private SettingsDocView generateSettingsDoc(
     return settingsDoc.build();
   }
 
+  private ApiAndSamples<StaticLangApiMethodView, StaticLangApiMethodView> methodAndSamplesFor(
```

Comment:
This function doesn't create a method, it only creates samples.

Overall you have woven together method & sample generation, which mixes and couples concerns. I think it would be better for sample generation to run outside of the context of client+method generation.

Reply:
You're right, it only creates the samples because the method is passed in. The reason I chose the name is that it also creates and returns the method+samples bundle `ApiAndSamples`

"you have woven together method & sample generation, which mixes and couples concerns": Previously I had client generation and then the sample creation based off that. I changed it response to your comment on the earlier pass: "I'm uneasy of the general approach here, where the client surface is generated first, and then you crack it back open as a basis for the samples. For all of the other stuff we generate, it all starts from the source data - this is done for the client, in-code samples, and tests. I'd prefer to have your samples work the same way instead of working differently."

"I think it would be better for sample generation to run outside of the context of client+method generation.": To some extent yes, and that was what my first pass was trying to do. That said, the samples should be derived from the methods insofar as makes sense: I don't want to replicate the logic in two places. That's why my first approach inspected the set of methods created by the library.

Reply:
ptal